### DISCUSSION TITLE
Best way to put simple content inside the node

### DISCUSSION THREAD

**Discussion by SciManJD** [bcakmakoglu/vue-flow#872] [Q&A] [answered]
I want to put a picture (png) inside the node. I found how can I do this in the documentation. I created a vue component and called it instead of the text in the "label". This works fine. But is there and easy way to do this without component just for one picture?

**Comment by bcakmakoglu** [comment, marked as answer]
What would be an "easier" way to do this for you?
Labels already allow you to pass a component instead of a simple string (which I assume you're using judging from your post), it won't get much easier than that, I'm afraid 😄

**Reply by SciManJD** [reply]
OK. I just want to be sure... I thought that I can use a sort of a slot to put my picture inside the node.

**Reply by bcakmakoglu** [reply]
To "simplify" passing an img, you could wrap your image in a render function

```ts
const img = () => h('img', { src: imgSrc, alt: 'some img' })

const elements = ref([
	{
		id: '1',
		position: { x: 0, y: 0 },
		label: img,
	}
])
```

**Reply by SciManJD** [reply]
Yes! This is exactly what I mean. But how to set a style? I tried this:
const img = () => h('img', { src: imgSrc, alt: 'some img', style: 'my style' })
but it doesn't work...

**Reply by bcakmakoglu** [reply]
`my style` isn't a valid style property though 😄 
If you want to add a *class name*, use `class: 'my-class'`
If you actually want to use inline styles, you can either use the object notation *or* the regular css sytnax

Object:
```ts
const img = () => h('img', { src: imgSrc, alt: 'some img', style: { backgroundColor: 'red' } })
```

CSS:
```ts
const img = () => h('img', { src: imgSrc, alt: 'some img', style: 'background-color: red' })
```

**Reply by SciManJD** [reply]
Aaah, OK. I made it with class: 'my-class' and it works just fine. This is even better than to use style. Thanks! No more questions concerning this topic!

**Reply by bcakmakoglu** [reply]
A quick tip: `h` supports the same syntax for `class` that Vue would in any other component, meaning you can also use conditional class names.

```ts
const someBool = ref(false)

setTimeout(() => {
	someBool.value = true
}, 2000)

const img = () => h('img', { src: imgSrc, alt: 'some img', class: { 'my-class': someBool.value } })
```

This would apply the `my-class` class name to the img tag whenever `someBool` is `true` :)
